When linking libtensorflow_cc.so.2 fails with many undefined references originating in grpc, e.g.:

bazel-out/k8-opt/bin/external/com_github_grpc_grpc/libtsi_ssl_credentials.pic.a(ssl_transport_security.pic.o):ssl_transport_security.cc:function populate_ssl_context(ssl_ctx_st*, tsi_ssl_pem_key_cert_pair const*, char const*): error: undefined reference to 'PEM_read_bio_PrivateKey'
bazel-out/k8-opt/bin/external/com_github_grpc_grpc/libtsi_ssl_credentials.pic.a(ssl_transport_security.pic.o):ssl_transport_security.cc:function populate_ssl_context(ssl_ctx_st*, tsi_ssl_pem_key_cert_pair const*, char const*): error: undefined reference to 'EVP_PKEY_free'

Fix by explicitely linking the dep that gets now ignored due to a change from 2.19, enabled by default in 2.20.

See https://github.com/tensorflow/tensorflow/pull/125250

Author: Alexander Grund (TU Dresden)

diff --git a/third_party/systemlibs/boringssl.BUILD b/third_party/systemlibs/boringssl.BUILD
--- a/third_party/systemlibs/boringssl.BUILD
+++ b/third_party/systemlibs/boringssl.BUILD
@@ -13,7 +13,7 @@ cc_library(
 
 cc_library(
     name = "ssl",
-    linkopts = ["-lssl"],
+    linkopts = ["-lssl", "-lcrypto"],
     visibility = ["//visibility:public"],
     deps = [
         ":crypto",
